### PR TITLE
support redirect url in oauth manager

### PR DESCRIPTION
**Pull request by lsdimagine**
This one is for using custom secure redirect url to skip allow page. I tested on paper iOS to see if we can get auth successfully.

### CODE REVIEW

File: Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBOAuthManager.m [hunk 1]
```diff
@@ -52,19 +52,27 @@ - (instancetype)initWithAppKey:(NSString *)appKey {
...
+    }
+
     _appKey = appKey;
-    _redirectURL = [[NSURL alloc] initWithString:[NSString stringWithFormat:@"db-%@://2/token", _appKey]];
+    _redirectURL = [[NSURL alloc] initWithString:redirectURL];
     _cancelURL = [NSURL URLWithString:[NSString stringWithFormat:@"db-%@://2/cancel", _appKey]];
     _host = host;
     _urls = [NSMutableArray arrayWithObjects:_redirectURL, nil];
-#ifdef TARGET_OS_X
+#ifdef TARGET_OS_MAC
```
Comment (scobbe) [inline Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBOAuthManager.m]: This needs to be `TARGET_OS_X`, I think

File: Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBOAuthManager.m [hunk 2]
```diff
@@ -52,19 +52,27 @@ - (instancetype)initWithAppKey:(NSString *)appKey {
 }
 
 - (instancetype)initWithAppKey:(NSString *)appKey host:(nullable NSString *)host {
+  return [self initWithAppKey:appKey host:host redirectURL:nil];
+}
+
+- (instancetype)initWithAppKey:(NSString *)appKey host:(nullable NSString *)host redirectURL:(nullable NSString *)redirectURL {
```
Comment (scobbe) [inline Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBOAuthManager.m]: Let's keep nullability keywords only in header file

Comment (lsdimagine) [reply Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBOAuthManager.m]: Any specific reason? It's called with nil redirectURL above.

File: Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBOAuthManager.m (hunk 1 above)
Comment (lsdimagine) [reply Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBOAuthManager.m]: Oops, fixed

**Comment by lsdimagine** [comment]
@scobbe Pushed the fix commit

File: Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBOAuthManager.m (hunk 2 above)
Comment (scobbe) [reply Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBOAuthManager.m]: It's redundant if it's in both header and impl file

Comment (lsdimagine) [reply Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBOAuthManager.m]: k that makes sense, removed.

File: Source/ObjectiveDropboxOfficial/Shared/Handwritten/Networking/DBTransportDefaultConfig.m [hunk 3]
```diff
@@ -105,16 +105,39 @@ - (instancetype)initWithAppKey:(NSString *)appKey
...
+              additionalHeaders:additionalHeaders
+                  delegateQueue:delegateQueue
+         forceForegroundSession:forceForegroundSession
+      sharedContainerIdentifier:sharedContainerIdentifier];
+}
+
+- (instancetype)initWithAppKey:(NSString *)appKey
+                     appSecret:(nullable NSString *)appSecret
+                hostnameConfig:(nullable DBTransportBaseHostnameConfig *)hostnameConfig
+                   redirectURL:(nullable NSString *)redirectURL
```
Comment (scobbe) [inline Source/ObjectiveDropboxOfficial/Shared/Handwritten/Networking/DBTransportDefaultConfig.m]: here too

Comment (lsdimagine) [reply Source/ObjectiveDropboxOfficial/Shared/Handwritten/Networking/DBTransportDefaultConfig.m]: Oops, sorry for missing them. Removed all nullable from this .m file.

**Comment by scobbe** [comment]
Thanks for making this change!

**Pull request merged**